src/main/scala/is/hail/distributedmatrix/BlockMatrixIsDistributedMatrix.scala

```diff
@@ -0,0 +1,244 @@
+package is.hail.distributedmatrix
+
+import is.hail.utils._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.mllib.linalg._
+import org.apache.spark.mllib.linalg.distributed._
+
+import scala.reflect.ClassTag
+
+object BlockMatrixIsDistributedMatrix extends DistributedMatrix[BlockMatrix] {
+  type M = BlockMatrix
+
+  def cache(m: M): M = m.cache()
+
+  def from(irm: IndexedRowMatrix, dense: Boolean = true): M =
+    if (dense) irm.toBlockMatrixDense()
+    else irm.toBlockMatrix()
+  def from(bm: BlockMatrix): M = bm
+  def from(cm: CoordinateMatrix): M =
```

Comment:
Coordinate matrices don't ever seem like a good idea to me.  If you don't use this anywhere (you don't, as far as I can see), I'd delete this.